File: utilcode/pkg/src/main/java/com/blankj/utilcode/pkg/feature/clean/CleanActivity.java
package com.blankj.utilcode.pkg.feature.clean;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.os.Environment;
import android.support.annotation.Nullable;
import android.view.View;

import com.blankj.utilcode.pkg.R;
import com.blankj.lib.base.BaseBackActivity;
import com.blankj.utilcode.util.CleanUtils;
import com.blankj.utilcode.util.SnackbarUtils;

import java.io.File;

/**
 * <pre>
 *     author: Blankj
 *     blog  : http://blankj.com
 *     time  : 2016/09/29
 *     desc  : demo about CleanUtils
 * </pre>
 */
public class CleanActivity extends BaseBackActivity {

    private View   snackBarRootView;
    private String internalCachePath;
    private String internalFilesPath;
    private String internalDbs;
    private String internalSp;
    private String externalCache;

    public static void start(Context context) {
        Intent starter = new Intent(context, CleanActivity.class);
        context.startActivity(starter);
    }

    @Override
    public void initData(@Nullable Bundle bundle) {

    }

    @Override
    public int bindLayout() {
        return R.layout.activity_clean;
    }

    @Override
    public void initView(Bundle savedInstanceState, View contentView) {
        setTitle(R.string.demo_clean);

        snackBarRootView = findViewById(android.R.id.content);
        findViewById(R.id.btn_clean_internal_cache).setOnClickListener(this);
        findViewById(R.id.btn_clean_internal_files).setOnClickListener(this);
        findViewById(R.id.btn_clean_internal_databases).setOnClickListener(this);
        findViewById(R.id.btn_clean_internal_sp).setOnClickListener(this);
        findViewById(R.id.btn_clean_external_cache).setOnClickListener(this);

        internalCachePath = getCacheDir().getPath();
        internalFilesPath = getFilesDir().getPath();
        internalDbs = getFilesDir().getParent() + File.separator + "databases";
        internalSp = getFilesDir().getParent() + File.separator + "shared_prefs";

        if (Environment.MEDIA_MOUNTED.equals(Environment.getExternalStorageState())) {
            externalCache = getExternalCacheDir().getAbsolutePath();
        }
    }

    @Override
    public void doBusiness() {

    }

    @Override
    public void onWidgetClick(View view) {
        int i = view.getId();
        if (i == R.id.btn_clean_internal_cache) {
            showSnackbar(CleanUtils.cleanInternalCache(), internalCachePath);

        } else if (i == R.id.btn_clean_internal_files) {
            showSnackbar(CleanUtils.cleanInternalFiles(), internalFilesPath);

        } else if (i == R.id.btn_clean_internal_databases) {
            showSnackbar(CleanUtils.cleanInternalDbs(), internalDbs);

        } else if (i == R.id.btn_clean_internal_sp) {
            showSnackbar(CleanUtils.cleanInternalSp(), internalSp);

        } else if (i == R.id.btn_clean_external_cache) {
            showSnackbar(CleanUtils.cleanExternalCache(), externalCache);

        }
    }

    private void showSnackbar(final boolean isSuccess, final String path) {
        if (isSuccess) {
            SnackbarUtils.with(snackBarRootView)
                    .setMessage("clean \"" + path + "\" dir success")
                    .setDuration(SnackbarUtils.LENGTH_LONG)
                    .showSuccess();
        } else {
            SnackbarUtils.with(snackBarRootView)
                    .setMessage("clean \"" + path + "\" dir failed")
                    .setDuration(SnackbarUtils.LENGTH_LONG)
                    .showError();
        }
    }
}

File: utilcode/pkg/src/main/java/com/blankj/utilcode/pkg/feature/clean/CleanActivity.kt
package com.blankj.utilcode.pkg.feature.clean

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.os.Environment
import android.view.View
import com.blankj.lib.base.BaseTitleBarActivity
import com.blankj.utilcode.pkg.R
import com.blankj.utilcode.util.CleanUtils
import com.blankj.utilcode.util.SnackbarUtils
import kotlinx.android.synthetic.main.activity_clean.*
import java.io.File

/**
 * ```
 * author: Blankj
 * blog  : http://blankj.com
 * time  : 2016/09/29
 * desc  : demo about CleanUtils
 * ```
 */
class CleanActivity : BaseTitleBarActivity() {

    companion object {
        fun start(context: Context) {
            val starter = Intent(context, CleanActivity::class.java)
            context.startActivity(starter)
        }
    }

    private lateinit var snackBarRootView: View
    private lateinit var internalCachePath: String
    private lateinit var internalFilesPath: String
    private lateinit var internalDbs: String
    private lateinit var internalSp: String
    private lateinit var externalCache: String

    override fun bindTitle(): CharSequence {
        return getString(R.string.demo_clean)
    }

    override fun initData(bundle: Bundle?) {}

    override fun bindLayout(): Int {
        return R.layout.activity_clean
    }

    override fun initView(savedInstanceState: Bundle?, contentView: View) {
        snackBarRootView = findViewById(android.R.id.content)
        cleanInternalCacheBtn.setOnClickListener(this)
        cleanInternalFilesBtn.setOnClickListener(this)
        cleanInternalDatabasesBtn.setOnClickListener(this)
        cleanInternalSpBtn.setOnClickListener(this)
        cleanExternalCacheBtn.setOnClickListener(this)

        internalCachePath = cacheDir.path
        internalFilesPath = filesDir.path
        internalDbs = filesDir.parent + File.separator + "databases"
        internalSp = filesDir.parent + File.separator + "shared_prefs"

        if (Environment.MEDIA_MOUNTED == Environment.getExternalStorageState()) {
            externalCache = externalCacheDir.absolutePath
        }
    }

    override fun doBusiness() {}

    override fun onWidgetClick(view: View) {
        when (view.id) {
            R.id.cleanInternalCacheBtn -> showSnackbar(CleanUtils.cleanInternalCache(), internalCachePath)
            R.id.cleanInternalFilesBtn -> showSnackbar(CleanUtils.cleanInternalFiles(), internalFilesPath)
            R.id.cleanInternalDatabasesBtn -> showSnackbar(CleanUtils.cleanInternalDbs(), internalDbs)
            R.id.cleanInternalSpBtn -> showSnackbar(CleanUtils.cleanInternalSp(), internalSp)
            R.id.cleanExternalCacheBtn -> showSnackbar(CleanUtils.cleanExternalCache(), externalCache)
        }
    }

    private fun showSnackbar(isSuccess: Boolean, path: String?) {
        SnackbarUtils.with(snackBarRootView)
                .setDuration(SnackbarUtils.LENGTH_LONG)
                .apply {
                    if (isSuccess) {
                        setMessage("clean \"$path\" dir success")
                        showSuccess()
                    } else {
                        setMessage("clean \"$path\" dir failed")
                        showError()
                    }
                }
    }
}
